### DISCUSSION TITLE
What would be needed to use Freya for Android ?

### DISCUSSION THREAD

**Discussion by felipetesc** [marc2332/freya#506] [General]
What type of things we might need to create a Freya app for Android ?

**Comment by marc2332** [comment]
Well, even though I have not tried to run Freya on Android, I guess you would use it to create GUI apps, what else could you use it for otherwise?

**Reply by anantnrg** [reply]
I think OP is asking **what** we need to use Freya on Android, not what we can use Freya for.

**Reply by marc2332** [reply]
Oh yeah, definitely, I missread it.  Some things that come to my mind:

- Prebuilt binaries of Skia for Android
- Probably adapt the event loop, I am pretty sure that there are a few things that would be needed to add or adapt for Android to work fine
- Integrate with Android APIs, like, opening/closing the keyboard

**Reply by anantnrg** [reply]
The only main thing that comes to mind is adding support for touch and drag events. I believe skia-safe has binaries for Android and iOS

**Reply by marc2332** [reply]
Freya already has touch events, and skia-safe doesn't have prebuilt binaries for the set of features it uses, but it's not a problem as long the maintainer would agree to adding android as a target.

But anyway, mobile support isn't something I am currently interested, but I don't discard I might add support in the future. I am happy to accept PRs tho

**Comment by felipetesc** [comment]
> Freya already has touch events, and skia-safe doesn't have prebuilt binaries for the set of features it uses, but it's not a problem as long the maintainer would agree to adding android as a target.
> 
> But anyway, mobile support isn't something I am currently interested, but I don't discard I might add support in the future. I am happy to accept PRs tho


Obviously the size of Freya apps will change for Android, but since rust compilation generates a very small binary size for desktops, if we use the flag  `opt-level = 'z'`, it could be very interesting to add support for Android/IOS. Sure, not right now, since as we noticed, you are doing a lot already. However, and because, there's a lot going on, I though it could be wise to bring this topic right now, so if changes are made there isn't the need to change so much in the future. 
For example, lets say you want to change the main loop, as you estated, bringing this subject right away may allow some changes to acomodate cellphone development  later.
